### PR TITLE
Added Bubblegum 96 Support

### PR DESCRIPTION
**Pull request by ProfFan**
This is my portings for the 96boards Bubblegum-96 board support. Tested on latest linaro build with i2c-dev.

### CODE REVIEW

File: src/arm/96boards.c [hunk 1]
```diff
@@ -34,23 +34,31 @@
...
-#define DT_BASE "/sys/firmware/devicetree/base"
+#define DT_BASE "/proc/device-tree"
 
 #define PLATFORM_NAME_DB410C "DB410C"
 #define PLATFORM_NAME_HIKEY "HIKEY"
+#define PLATFORM_NAME_BBGUM "BBGUM"
 
 int db410c_ls_gpio_pins[MRAA_96BOARDS_LS_GPIO_COUNT] = {
-  36, 12, 13, 69, 115, 4, 24, 25, 35, 34, 28, 33,
+   36, 12, 13, 69, 115, 4, 24, 25, 35, 34, 28, 33,
```
Comment (alext-mkrs) [inline src/arm/96boards.c]: Whitespace error at the beginning of the string

File: src/arm/96boards.c [hunk 2]
```diff
@@ -34,23 +34,31 @@
...
 };
 
 const char* db410c_serialdev[MRAA_96BOARDS_LS_UART_COUNT] = { "/dev/ttyMSM0", "/dev/ttyMSM1"};
 
 int hikey_ls_gpio_pins[MRAA_96BOARDS_LS_GPIO_COUNT] = {
- 488, 489, 490, 491, 492, 415, 463, 495, 426, 433, 427, 434,
+   488, 489, 490, 491, 492, 415, 463, 495, 426, 433, 427, 434,
```
Comment (alext-mkrs) [inline src/arm/96boards.c]: Whitespace error at the beginning of the string

File: src/arm/96boards.c [hunk 3]
```diff
@@ -34,23 +34,31 @@
...
 };
 
+
```
Comment (alext-mkrs) [inline src/arm/96boards.c]: Surplus stying edit

File: src/arm/96boards.c [hunk 4]
```diff
@@ -34,23 +34,31 @@
 #include "common.h"
 #include "arm/96boards.h"
 
-#define DT_BASE "/sys/firmware/devicetree/base"
+#define DT_BASE "/proc/device-tree"
```
Comment (alext-mkrs) [inline src/arm/96boards.c]: [Disclaimer: I don't have any of these boards] Are you sure this won't break other boards supported by this file? Is that the case that this path is now `/proc/device-tree` on all those boards and their respective distros/versions?

File: src/arm/96boards.c [hunk 5]
```diff
@@ -97,7 +105,12 @@ mraa_board_t* mraa_96boards()
             ls_gpio_pins = hikey_ls_gpio_pins;
             b->uart_dev[0].device_path = hikey_serialdev[0];
             b->uart_dev[1].device_path = hikey_serialdev[1];
-	}
+	     } else if (mraa_file_contains(DT_BASE "/model", "s900")) {
```
Comment (alext-mkrs) [inline src/arm/96boards.c]: This should be on a separate line (hint: run `clang-format` on your changes using config file from the root of the repo)

File: src/arm/96boards.c (hunk 4 above)
Comment (ProfFan) [inline src/arm/96boards.c]: I have not tried these boards, but this path is incorrect itself. As stated by the Linux ABI, (https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-firmware-ofw),
>  Userspace must not use the `/sys/firmware/devicetree/base` path directly, but instead should follow `/proc/device-tree` symlink. It is possible that the absolute path will change in the future, but the symlink is the stable ABI.

I believe using this path is a common misconception in the community.

File: src/arm/96boards.c (hunk 1 above)
Comment (ProfFan) [inline src/arm/96boards.c]: Will fix ASAP.

**Comment by ProfFan** [comment]
I have not tried these boards, but this path is incorrect itself. As stated by the Linux ABI, (https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-firmware-ofw),
>  Userspace must not use the `/sys/firmware/devicetree/base` path directly, but instead should follow `/proc/device-tree` symlink. It is possible that the absolute path will change in the future, but the symlink is the stable ABI.

I believe using this path is a common misconception in the community.

All source files have been `clang-format`ed.

Added `Signed-off-by`.

BTW, I have a bunch other 96boards types on the way. Will add them when they arrive 😄

**Comment by arfoll** [comment]
This is great, thanks alot! Merged.

**Pull request closed without merging**